alexandria/core/tests/test_views.py

```diff
@@ -201,40 +201,57 @@ def test_hook_view(
 
 
 @pytest.mark.parametrize(
-    "admin_groups, active_group, expect_failure",
+    "admin_groups, update, active_group, expect_response",
     [
-        (["foo", "bar"], "foo", False),
-        (["foo", "bar"], None, False),
-        (["foo", "bar"], "somethingelse", True),
+        (["foo", "bar"], False, "foo", HTTP_201_CREATED),
+        (["foo", "bar"], False, None, HTTP_201_CREATED),
+        (["foo", "bar"], False, "somethingelse", HTTP_400_BAD_REQUEST),
+        (["foo", "bar"], True, "foo", HTTP_200_OK),
+        (["somegroup"], True, "foo", HTTP_200_OK),
+        (["somegroup"], True, "somegroup", HTTP_200_OK),
+        (["foobar"], True, "somegroup", HTTP_200_OK),
     ],
 )
 @pytest.mark.parametrize("viewset", [DocumentViewSet, FileViewSet, TagViewSet])
 def test_validate_created_by_group(
-    db, viewset, request, admin_client, active_group, expect_failure
+    db, viewset, request, update, admin_client, active_group, expect_response,
 ):
     viewset_inst = viewset()
     model_name = viewset_inst.queryset.model.__name__.lower()
-    model = request.getfixturevalue(f"{model_name}_factory").create()
+    model = request.getfixturevalue(f"{model_name}_factory").create(
+        created_by_group="somegroup"
+    )
     serializer = viewset_inst.serializer_class()
     serialized_model = serializer.to_representation(model)
-
-    # delete model in the DB so we can then create it via API
-    model.delete()
     serialized_model["created_by_group"] = active_group
+    del serialized_model["modified_by_group"]
+
+    if not update:
+        # delete model in the DB so we can then create it via API
+        model.delete()
 
     post_data = {
         # Note: model name pluralisation may not always be with an "s" suffix,
         # but for now, this holds true
         "data": {"attributes": serialized_model, "type": f"{model_name}s"}
```

Comment:
You could use `model.__class__._meta.verbose_name_plural` in order to always have the same plural name as rest framework json api.

Reply:
Done, good point!